### PR TITLE
chore: complete more incomplete types

### PR DESCRIPTION
**Pull request by henryiii**
- chore: add some incomplete types
- chore: modernize pyproject.toml

### Description

Adding more incomplete types. About 2/3 of the way through being able to turn on the strictness flag for it, but traveling tomorrow so just going with what I have now, and can do more later.

### CODE REVIEW

File: packages/cryptography/test_cryptography.py [hunk 1]
```diff
@@ -76,7 +76,7 @@ def test_der():
             raise ZeroDivisionError
 
     # Parse the outer element.
-    outer = reader.read_element(SEQUENCE)  # type: ignore[unreachable]
+    outer = reader.read_element(SEQUENCE)
```
Comment (henryiii) [inline packages/cryptography/test_cryptography.py:79]: Nice side affect of adding pytest to typing.

File: pyodide-build/pyodide_build/mkpkg.py [hunk 2]
```diff
@@ -48,7 +48,7 @@ def _find_wheel(pypi_metadata: dict[str, Any]) -> dict[str, Any] | None:
 
 
 def _find_dist(
-    pypi_metadata: dict[str, Any], source_types=list[Literal["wheel", "sdist"]]
+    pypi_metadata: dict[str, Any], source_types: list[Literal["wheel", "sdist"]]
```
Comment (henryiii) [inline pyodide-build/pyodide_build/mkpkg.py:51]: This was "missing" a type ;)

File: src/py/pyproject.toml [hunk 3]
```diff
@@ -1,2 +1,3 @@
 [build-system]
-requires = ["setuptools>=42", "wheel"]
+requires = ["setuptools>=42"]
+build-backend = "setuptools.build_meta"
```
Comment (henryiii) [inline src/py/pyproject.toml:3]: Fixed this up - wheel is not needed (declared by setuptools via PEP 517), and you should always specify a build-backend.

File: pyodide-build/pyodide_build/buildall.py [hunk 4]
```diff
@@ -126,7 +126,7 @@ def tests_path(self) -> Path | None:
             return tests[0]
         return None
 
-    def build(self, outputdir: Path, args) -> None:
+    def build(self, outputdir: Path, args: Any) -> None:
```
Comment (hoodmane) [inline pyodide-build/pyodide_build/buildall.py:129]: Maybe
```suggestion
    def build(self, outputdir: Path, args: argparse.Namespace) -> None:
```

**Review by hoodmane — APPROVED** [review]

**Comment by hoodmane** [comment]
Thanks @henryiii!

**Pull request merged**